api/src/main/java/org/openmrs/module/sync2/api/scheduler/SyncPullTask.java

```diff
@@ -0,0 +1,40 @@
+package org.openmrs.module.sync2.api.scheduler;
+
+import org.openmrs.api.context.Context;
+import org.openmrs.module.sync2.client.reader.ParentFeedReader;
+import org.openmrs.module.sync2.client.reader.impl.ParentFeedReaderImpl;
+import org.openmrs.scheduler.tasks.AbstractTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SyncPullTask extends AbstractTask {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SyncPullTask.class);
+
+    private ParentFeedReader parentFeedReader;
+
+    /**
+     * @see org.openmrs.scheduler.tasks.AbstractTask#execute()
+     */
+    @Override
+    public void execute() {
+        parentFeedReader = Context.getRegisteredComponent("sync2.parentFeedReader", ParentFeedReaderImpl.class);
+
+        if (!isExecuting) {
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
I think that this check is redundant. Logger manages log levels on it's own I think (correct me if I'm wrong please). Moreover I think that executing sync task should be logged as *info* level.